frontend/src/App.jsx

```diff
@@ -14,48 +14,63 @@ import { API_BASE_URL } from './config.js';
 
 function App() {
   const [user, setUser] = useState(null);
-  const [isLoading, setIsLoading] = useState(true);
+  const [flightData, setFlightData] = useState(initialFlightData);
+  const { isLoading, setIsLoading } = useLoading();
+  const [error, setError] = useState(null);
 
-  const updateUser = (newUser) => {
-    setUser(newUser);
+  const fetchUserData = async () => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(`${API_BASE_URL}/users/data`, {
+        credentials: 'include',
+      });
+      if (response.ok) {
+        const userData = await response.json();
+        updateUser(userData);
+      } else {
+        setUser(null);
+      }
+    } catch (error) {
+      setError('Error fetching user data:', error);
+      setUser(null);
+    } finally {
+      setIsLoading(false);
+    }
   };
 
   useEffect(() => {
-    //making a call to backend middleware API to check the authentication status of user
-    const checkAuth = async () => {
-      try {
-        const response = await fetch(`${API_BASE_URL}/users/auth-check`, {
-          credentials: 'include'
-        });
-        if (response.ok) {
-          const data = await response.json();
-          setUser(data.user);
-        }
-      } catch (error) {
-        console.error('Auth check failed:', error);
-      } finally {
-        setIsLoading(false);
-      }
-    };
-    checkAuth();
-  }, []);
+    fetchUserData();
+  },[]);
 
-  if (isLoading) {
-    return <div>Loading...</div>;
-  }
+  const updateUser = (newUser) => {
+    setUser(newUser);
+  };
+
+  const updateFlightData = (newData) => {
+    setFlightData(prevData => ({ ...prevData, ...newData }));
+  };
 
   
   return (
     <div className="app">
-      <UserContext.Provider value={{ user, updateUser }}>
+      <UserContext.Provider value={{ user, updateUser, flightData, updateFlightData, fetchUserData }}>
         <LoadingProvider>
           <BrowserRouter>
             <Routes>
-              <Route path="/" element={user ? <HomePage/>: <LoginForm />} />
+              {isLoading ? (
```

Comment:
instead of doing this. How about creating a specific page with a loader if we want to go that route and redirect to it https://reactrouter.com/en/main/fetch/redirect . Alternatively showing a loader component on the active page would look nicer.